File: libaums/src/main/java/com/github/mjdev/libaums/fs/Fat32FileSystemCreator.java
package com.github.mjdev.libaums.fs;

import com.github.mjdev.libaums.driver.BlockDeviceDriver;
import com.github.mjdev.libaums.fs.fat32.Fat32FileSystem;
import com.github.mjdev.libaums.partition.PartitionTableEntry;
import com.github.mjdev.libaums.partition.PartitionTypes;

import java.io.IOException;

/**
 * Created by magnusja on 28/02/17.
 */

public class Fat32FileSystemCreator implements FileSystemCreator {

    @Override
    public FileSystem read(PartitionTableEntry entry, BlockDeviceDriver blockDevice) throws IOException {
        if(entry.getPartitionType() != PartitionTypes.FAT32) return null;

        return Fat32FileSystem.read(blockDevice);
    }
}

File: libaums/src/main/java/me/jahnen/libaums/fs/fat32/Fat32FileSystemCreator.kt
package me.jahnen.libaums.fs.fat32

import me.jahnen.libaums.driver.BlockDeviceDriver
import me.jahnen.libaums.fs.FileSystem
import me.jahnen.libaums.fs.FileSystemCreator
import me.jahnen.libaums.partition.PartitionTableEntry
import java.io.IOException

/**
 * Created by magnusja on 28/02/17.
 */

class Fat32FileSystemCreator : FileSystemCreator {

    @Throws(IOException::class)
    override fun read(entry: PartitionTableEntry, blockDevice: BlockDeviceDriver): FileSystem? {
        return Fat32FileSystem.read(blockDevice)
    }
}
